action.php

```diff
@@ -144,6 +158,34 @@ function preprocess(&$event, $param) {
         }
     }
 
+    /**
+     * process login action
+     *
+     * @param obj $event
+     * @param array $param
+     *
+     */
+    public function login_preprocess(Doku_Event $event, $param) {
+        global $INPUT;
+        if(!$this->getConf('loginprotect')) return; // no protection wanted
+        if(!$INPUT->bool('u')) return; // this login was not triggered by a form
+
+        // we need to have $ID set for the captcha check
+        global $ID;
+        $ID = getID();
+
+        /** @var helper_plugin_captcha $helper */
+        $helper = plugin_load('helper', 'recaptcha2');
+        $resp = $helper->check();
+
+        if(!$resp->isSuccess()) {
+            $event->data['silent'] = true; // we have our own message
+            $event->result = false; // login fail
+            $event->preventDefault();
+            $event->stopPropagation();
```

Comment:
If the login fails, the user should get an error-message, telling him what happened.

You can use the same mechanism as the other checks, that just throw a `msg($this->getLang('testfailed'),-1);`